### DISCUSSION TITLE
blocked websocket / delayed messages?

### DISCUSSION THREAD

**Discussion by louisholley** [enisdenjo/graphql-ws#309] [Q&A] [answered]
not entirely sure if this is a graphql-ws issue but I'm really struggling to figure out what's going on and don't know where else to seek advice...

I'm using graphql-ws for a messaging platform, and I have two identical urql clients set up - one in a nextjs app and one in a react native app. I've got both clients connected to the server and I'm testing having a conversation between the two. The situation is as follows:
- when sending messages from the next app, they are received immediately by the react native app
- when sending messages from the react native app, they are not received by the next app, until another message is sent by the next app. only once this message is sent is the preceding message received. these screenshots visualise the behaviour (note the order)...

conversation according to the next app:
<img width="820" alt="image" src="https://user-images.githubusercontent.com/4563781/154689992-35b6fc5d-4f5f-4bff-be9f-2a0ef328a62c.png">

conversation according to the react native app:
<img width="259" alt="image" src="https://user-images.githubusercontent.com/4563781/154690025-8f3da494-215d-4805-9ac7-6535f428a233.png">

what doesn't make sense to me is that the urql clients are setup in exactly the same way, and the react native app is receiving the messages fine without this weird blocking/delay. our backend is severless and the messages are enqueued and published to subscribers in the correct order as the react native app screenshot shows, so it seems unlikely to me that this could be a server issue. I'm wondering if anyone has ever experienced anything like this and has any advice wrt debugging and resolving this issue? I don't even know where to start! hoping this is allowed as I'm completely in the dark as to how to proceed...

thanks so much in advance!

**Comment by louisholley** [comment]
it's worth adding: an even weirder thing is that the mutation to send the message uses a separate client to the subscription client, so it's baffling that the subscription client only receives its next message when these mutations happen

**Comment by enisdenjo** [comment, marked as answer]
I doubt `graphql-ws` is causing this, so I converted the issue to a discussion. Still, it's interesting, so let's investigate further.

I'll start by asking: have you checked the messages themselves? Are there messages sent to/by `graphql-ws`?

On the server, please consult the ["server usage with console logging" recipe](https://github.com/enisdenjo/graphql-ws#logging).

For the client, you can [listen on message events](https://github.com/enisdenjo/graphql-ws/blob/master/docs/modules/client.md#eventmessagelistener) and log the results to confirm.

```ts
import { createClient } from 'graphql-ws';

const client = createClient({
  ...
  on: {
    message: (msg) => {
      console.log(msg);
    },
  },
});
```

**Reply by louisholley** [reply]
thanks so much for your quick reply  - I redeployed the server with these logs and the issue disappeared 🤦...will definitely be using these in the future if it happens again, and I'll keep you updated if so!
